src/XMakeBuildEngine/Construction/ProjectRootElement.cs

```diff
@@ -1714,8 +1714,23 @@ public void Save(Encoding saveEncoding)
                 {
                     using (ProjectWriter projectWriter = new ProjectWriter(_projectFileLocation.File, saveEncoding))
                     {
-                        projectWriter.Initialize(XmlDocument);
-                        XmlDocument.Save(projectWriter);
+                        var xmlWithNoImplicits = (XmlDocument)XmlDocument.Clone();
+
+                        var implicitElements =
+                            xmlWithNoImplicits.SelectNodes($"//*[@{XMakeAttributes.@implicit}]");
+
+                        if (implicitElements == null)
```

Comment:
I did some searching and couldn't find an instance where it would return null unless the document couldn't be parsed.  I think its safe to remove the check personally.